content/posts/Learning-Passport.md

```diff
@@ -0,0 +1,485 @@
+---
+title: "Authenticating Node.js Applications using Passport"
+date: 2018-07-13T10:00:49+05:30
+draft: false
+author: "Prince Mittal"
+cover_image: "/images/passport/passport.png"
+aliases:
+    - /posts/asynchronousjs/
+---
+
+### What is Passport.js? 
+
+As we all know, User authentication is fundamental to almost every web application in existence.Now given the ubiquity of the need to impelement this user authentication as a feature of our web apps, it wouldn't make sense to implement it from scratch each time.This is where passport comes in.
+
+[Passport](http://www.passportjs.org/) is a authentication middleware for Node.js applications. It's extremely flexible and modular, that can be fully customised and works great with express web applications. It is flexible in the sense that it allows for different authentication strategies like Local Strategy(username and password), Facebook, Twitter and more.
+
+
+All of the code from this article is available for download on [GitHub](https://github.com/mittalprince/Everyday-Stuff/tree/master/Blog%20Folder/learn_passport).
+
+### Prerequisites
+
+* __Node.js__
+	 	- Download and install [Node.js](https://nodejs.org/en/)
+
+* __MongoDB__
+		- Download and install [MongoDB Community Server](https://www.mongodb.com/download-center#atlas).Follow the instructions for your OS. Note, if you’re using Ubuntu, this [guide](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/) can help you get Mongo up and running.
+
+
+### Creating the Project
+
+Our Project structure and dependencies ``` (package.json) ``` will look like :
+
+<div class="row">
+        <img class="responsive-img col s10" src="/images/passport/project_structure.png">
+    </div>
+
+
+Once all the prerequisites software is set up, we can start our project. First create the folder for app and open the terminal and write the following command for node app.
+
+```
```

Comment:
change 
```
\`\`\`
    npm init
\`\`\`
```
to
```
\`\`\` bash
    npm init
\`\`\`
```
Language should be mentioned after \`\`\` and for command line use 'bash'. Do similar changes for other commands.